### PR TITLE
Add tests for the Side Panel Toolbar Button toggling

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Follow up for https://github.com/brave/brave-core/pull/15062

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/sidebar/BUILD.gn [hunk 1]
```diff
@@ -65,14 +65,18 @@ source_set("browser_tests") {
     testonly = true
     defines = [ "HAS_OUT_OF_PROC_TEST_RUNNER" ]
 
-    sources = [ "sidebar_browsertest.cc" ]
+    sources = [
+      "//brave/browser/ui/views/sidebar/sidebar_container_view_browsertest.cc",
```
Comment (fallaciousreasoning) [inline browser/ui/sidebar/BUILD.gn:69]: Not sure if this is the right place to add this.

**Comment by sangwoo108** [comment]
Hi, @fallaciousreasoning ! Is this needed to be rebased onto the latest main branch?

**Comment by fallaciousreasoning** [comment]
Hmm, weird, I would've thought it would detect those other commits got merged in. Rebased, hopefully the diff is a bit more sane now.

File: browser/ui/views/sidebar/sidebar_container_view_browsertest.cc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/brave_browser.h"
+#include "brave/browser/ui/sidebar/sidebar_controller.h"
+#include "brave/browser/ui/sidebar/sidebar_service_factory.h"
+#include "brave/browser/ui/views/frame/brave_browser_view.h"
+#include "brave/browser/ui/views/sidebar/sidebar_button_view.h"
+#include "brave/browser/ui/views/sidebar/sidebar_container_view.h"
```
Comment (sangwoo108) [inline browser/ui/views/sidebar/sidebar_container_view_browsertest.cc:11]: Please move up this header to the first line with empty line. I believe it's upstream convention to locate "foo.h" above all others in "foo_test.cc".

```suggestion
#include "brave/browser/ui/views/sidebar/sidebar_container_view.h"

#include "brave/browser/ui/brave_browser.h"
#include "brave/browser/ui/sidebar/sidebar_controller.h"
#include "brave/browser/ui/sidebar/sidebar_service_factory.h"
#include "brave/browser/ui/views/frame/brave_browser_view.h"
#include "brave/browser/ui/views/sidebar/sidebar_button_view.h"
```

File: browser/ui/views/sidebar/sidebar_container_view_browsertest.cc [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+#include "chrome/browser/ui/views/frame/toolbar_button_provider.h"
+#include "chrome/browser/ui/views/toolbar/side_panel_toolbar_button.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+class SidebarContainerViewBrowserTest : public InProcessBrowserTest {
+ public:
+  SidebarContainerViewBrowserTest() {}
```
Comment (sangwoo108) [inline browser/ui/views/sidebar/sidebar_container_view_browsertest.cc:25]: optional: 
```suggestion
  SidebarContainerViewBrowserTest() = default;
```

File: browser/ui/views/sidebar/sidebar_container_view_browsertest.cc [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  SidebarContainerViewBrowserTest(const SidebarContainerViewBrowserTest&) =
+      delete;
+  SidebarContainerViewBrowserTest& operator=(
+      const SidebarContainerViewBrowserTest&) = delete;
+
```
Comment (sangwoo108) [inline browser/ui/views/sidebar/sidebar_container_view_browsertest.cc:30]: nit: d'tor for a child class

```suggestion
  ~SidebarContainerViewBrowserTest() override = default;
```

File: browser/ui/views/sidebar/sidebar_container_view_browsertest.cc [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        ->toolbar_button_provider()
+        ->GetSidePanelButton();
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(SidebarContainerViewBrowserTest,
+                       ButtonIsShownByDefault) {
+  EXPECT_LT(0u, GetService()->items().size());
+  EXPECT_NE(nullptr, sidebar());
+  EXPECT_NE(nullptr, toolbar_button());
```
Comment (sangwoo108) [inline browser/ui/views/sidebar/sidebar_container_view_browsertest.cc:57]: nit: maybe simpler?

```suggestion
  EXPECT_TRUE(sidebar());
  EXPECT_TRUE(toolbar_button());
```

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

**Comment by fallaciousreasoning** [comment]
Thanks for the tips @sangwoo108!

**Pull request merged**